source/material_model/rheology/strain_dependent.cc

```diff
@@ -283,6 +308,35 @@ namespace aspect
         friction_strain_weakening_factors = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Friction strain weakening factors"))),
                                                                                     n_fields,
                                                                                     "Friction strain weakening factors");
+
+        if (prm.get ("Strain healing mechanism") == "no healing")
+          healing_mechanism = no_healing;
+        else if (prm.get ("Strain healing mechanism") == "temperature dependent")
+          healing_mechanism = temperature_dependent;
+        else
+          AssertThrow(false, ExcMessage("Not a valid Strain healing mechanism!"));
+
+        // Currently this functionality only works in field composition
+        if (healing_mechanism != no_healing)
```

Comment:
You are right, we first need to check if there are particles. This condition needs to be extended in the following way:

```suggestion
        if (healing_mechanism != no_healing && this->get_postprocess_manager().template has_matching_postprocessor<Postprocess::Particles<dim> >())
```